components/abstractions/src/test/java/com/microsoft/kiota/authentication/ApiKeyAuthenticationProviderTest.java

```diff
@@ -20,34 +20,32 @@ public void DefensivePrograming() {
 	}
 
 	@Test
-	public void AddsInQueryParameter() throws IllegalStateException, URISyntaxException {
+	void AddsInQueryParameter() throws IllegalStateException, URISyntaxException {
 		var value = new ApiKeyAuthenticationProvider("key", "param", ApiKeyLocation.QUERY_PARAMETER);
 		var request = new RequestInformation() {{
 			urlTemplate = "https://localhost{?param1}";
 		}};
-		value.authenticateRequest(request, null).join();
-		assertNull(request.headers.get("param"));
+		value.authenticateRequest(request, null);
 		assertEquals("https://localhost?param=key", request.getUri().toString());
 	}
 
 	@Test
-	public void AddsInQueryParameterWithOtherParameters() throws IllegalStateException, URISyntaxException {
+	void AddsInQueryParameterWithOtherParameters() throws IllegalStateException, URISyntaxException {
 		var value = new ApiKeyAuthenticationProvider("key", "param", ApiKeyLocation.QUERY_PARAMETER);
 		var request = new RequestInformation() {{
 			urlTemplate = "https://localhost{?param1}";
 		}};
 		request.addQueryParameter("param1", "value1");
-		value.authenticateRequest(request, null).join();
-		assertNull(request.headers.get("param"));
+		value.authenticateRequest(request, null);
```

Comment:
No reason to drop the assert
```suggestion
		value.authenticateRequest(request, null);
		assertNull(request.headers.get("param"));
```